tools/cli/src/mentat_cli/command_parser.rs

```diff
@@ -0,0 +1,196 @@
+// Copyright 2017 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use combine::{eof, many, many1, sep_by, skip_many, token, Parser};
+use combine::combinator::{choice, try};
+use combine::char::{alpha_num, space, string};
+
+pub static HELP_COMMAND: &'static str = &"help";
+pub static OPEN_COMMAND: &'static str = &"open";
+pub static CLOSE_COMMAND: &'static str = &"close";
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub enum Command {
+    Transact(Vec<String>),
+    Query(Vec<String>),
+    Help(Vec<String>),
+    Open(String),
+    Close,
+    Err(String),
+}
+
+impl Command {
+    pub fn is_complete(&self) -> bool {
+        match self {
+            &Command::Query(_) |
+            &Command::Transact(_) => false,
+            _ => true
```

Comment:
Enumerate every command rather than using `_` — that way this'll fail to compile if you add a new command and forget to extend this `match`!